src/Raven.Studio/typescript/viewmodels/database/status/storageReport.ts

```diff
@@ -172,28 +174,74 @@ class storageReport extends viewModelBase {
         };
         return preAllocatedBuffers;
     }
+    
+    private groupChildren(items: storageReportItem[]): storageReportItem[] {
+        
+        if (items.length > storageReport.maxChildrenToShow) {
+            let clonedItems = _.cloneDeep(items);
+            clonedItems.sort((a, b) => b.size - a.size); // desc...
+            
+            const sampleItem = clonedItems[0];
+            
+            const itemsToShow = clonedItems.slice(0, storageReport.maxChildrenToShow);
+            const itemsToGroup = clonedItems.slice(storageReport.maxChildrenToShow);
+            
+            const groupedItem: storageReportItem = itemsToGroup.reduce((a, b) => {
+                    a.size += b.size;
+                    a.length += b.length;
+                    a.pageCount += b.pageCount;
+                    a.numberOfEntries += b.numberOfEntries;
+                    return a;
+                },
+                {
+                    isGrouped: true,
+                    name: "Grouped Data",
+                    size: 0,
+                    length: 0,
+                    pageCount: 0,
+                    numberOfEntries: 0,
+                    internalChildren: [],
+                    hasChildren: () => false,
+                    lazyLoadChildren: false,
+                    type: sampleItem.type,
+                    parent: sampleItem.parent,
+                    showType: sampleItem.showType,
+                    formatSize: sampleItem.formatSize,
+                    formatPercentage: sampleItem.formatPercentage,
+                    recyclableJournal: sampleItem.recyclableJournal,
+                    customSizeProvider: sampleItem.customSizeProvider,
+                    isStorageEnvironment: sampleItem.isStorageEnvironment
+                })
```

Comment:
this looks very suspicious - what do you want to achieve here?

Reply:
just needed to provide an initial `storageReportItem `object for the reduce